### ISSUE TITLE
[Godot4.0] Provide a bool argument for  `Object.get_property_list()` and `Script.get_script_property_list()`, to decide exclude editor properties or not.

### ISSUE THREAD

**Issue by Daylily-Zeleen** [godotengine/godot-proposals#5806]
### Describe the project you are working on

Developing a tool which need serialize and deserialize.

### Describe the problem or limitation you are having in your project

For convenient, I using `Script.get_script_property_list()` to get all my custom properties, so I need not to code every property of every script for serialize.

But the return array has some properties which only using in editor and should not be considered as properties of a object, for example, Categories.

This will not happen in 3.x.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I already implement this feature [#68780](https://github.com/godotengine/godot/issues/68780):
![image](https://user-images.githubusercontent.com/61624558/202616996-e7a4d06b-452d-4773-bc94-1e521048859c.png)

Here is the test result:
![image](https://user-images.githubusercontent.com/61624558/202624901-bdc5886e-c366-4d89-af72-17aa9e92128d.png)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

1. Implement `void Object::_exclude_editor_properties(List<PropertyInfo> *p_list) const` to excute exclude behavior.
2. Add a argument to `Object::_get_property_list_bind()` and `Script::_get_script_property_list()`:
    `TypedArray<Dictionary> Object::_get_property_list_bind(bool without_editor_properties) const`
    `TypedArray<Dictionary> Script::_get_script_property_list(bool without_editor_properties)`
    , then implement them.

The default value of `without_editor_properties` is `false`, so it will not be a break change.
I think it should be `true` as default, commonly we just want to get properties by using `Object.get_property_list()` and  `Script.get_script_property_list()`, just like in 3.x. 
This need disscussion.

But this plan can not filter properties clearly, for example:
The property 'script', it should not be a property of `Object`, because a specific object behavior is define by a script. 
In other words, a script let object become a specific type object.
We can using `obj.set_script(script)` method to change a object to another type of object, but we should not using `obj.script = script` to change.
However, I think the usage of properties should be an another discussion. 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Of course, we can filter properies which we need, but obviously, these two methods' return value is inconsistent with their method name.

### Is there a reason why this should be core and not an add-on in the asset library?

This is about improving core.

**Comment by YuriSizov** [comment]
I'll close this and the other proposal so that the discussion of your issue can stay focused in one place (you've now opened 1 issue, 2 proposals, and 1 PR, which makes it impossible for people to keep track of what is being said). We can reopen it later if there is a conclusion for this.

For now, refer to https://github.com/godotengine/godot/issues/68780.
